python/dgl/graph_index.py

```diff
@@ -536,23 +537,26 @@ def node_subgraphs(self, vs_arr):
             gis.append(self.node_subgraph(v))
         return gis
 
-    def edge_subgraph(self, e):
+    def edge_subgraph(self, e, preserve_nodes=False):
         """Return the induced edge subgraph.
 
         Parameters
         ----------
         e : utils.Index
             The edges.
+        preserve_nodes: bool
+            Indicates whether to preserve all nodes or not. (when set to false, all nodes without edge connected to it would be removed.)
```

Comment:
If true, keep the nodes which have no edge connected in the subgraph.
It is equivalent to a sparser graph that contains only the given edges.